neurom/apps/morph_stats.py

```diff
@@ -73,34 +82,77 @@ def _stat_name(feat_name, stat_mode):
     return '%s_%s' % (stat_mode, feat_name)
 
 
-def extract_stats(neurons, config):
+def extract_stats(neurons, config, as_frame=False):
+    """Extract stats from neurons.
+
+    Arguments:
+        neurons: a neuron, population or neurite tree
+        config (dict): configuration dict. The keys are:
+            - neurite_type: a list of neurite types for which features are extracted
+            - neurite: a dictionary {neurite_feature: mode} where:
+                - neurite_feature is a string from NEURITEFEATURES
+                - mode is an aggregation operation provided as a string such as:
+                  ['min', 'max', 'median', 'mean', 'std', 'raw', 'total']
+        as_frame (bool): if yes, returns a dataframe, else a dict
+
+    Returns:
+        The extracted statistics
+
+    Note:
+        An example config can be found at:
+    """
+    if as_frame:
+        return _extract_stats_as_frame(neurons, config)
+    return _extract_stats_as_dict(neurons, config)
+
+
+extract_stats.__doc__ = extract_stats.__doc__ + EXAMPLE_CONFIG + '\n'
+
+
+def _extract_stats_as_frame(neurons, config):
+    del config['neuron']
+    stats = {nrn.name: extract_stats(nrn, config) for nrn in neurons}
```

Comment:
If I give a single neuron as input, it fails here

Reply:
also, I use neurom.load_neurons(list_of_paths), and the neuron.name=None, so I only get the last neuron in the dataframe

Reply:
I did that:
```python
    nrns = nm.load_neurons([os.path.join(SWC_PATH, name)
                            for name in ['Neuron.swc', 'simple.swc']])
    print(ms.extract_dataframe(nrns, REF_CONFIG))
```

and the output is:
```
     name     neurite_type  ...  max_segment_midpoint_Y  max_segment_midpoint_Z
0  Neuron             axon  ...                0.000000               49.520306
1  Neuron  apical_dendrite  ...                0.000000               53.750948
2  Neuron   basal_dendrite  ...               48.481977               51.575581
3  Neuron              all  ...               48.481977               53.750948
4  simple             axon  ...               -2.000000                0.000000
5  simple  apical_dendrite  ...                     NaN                     NaN
6  simple   basal_dendrite  ...                5.000000                0.000000
7  simple              all  ...                5.000000                0.000000
```

Do you know how did you end up with neuron.name = None ?